grades/tests.py

```diff
@@ -1,3 +1,16 @@
-# from django.test import TestCase
+import pytest
+from django.db.models.query import QuerySet
+from grades.models import Grade
 
-# Create your tests here.
+
+@pytest.mark.django_db
+class TestGrade:
+    def test_addGradeUserCourse(cls):
+        out = Grade.objects.all()
+        assert isinstance(out, QuerySet)
+        assert all(isinstance(g, Grade) for g in out)
+        assert list(out.values_list('user', 'course', 'grade', 'semester', 'year')) == [
+            (1, 1, 85, 'winter', 2021),
+            (1, 2, 60, 'spring', 2020),
+            (1, 1, 20, 'summer', 2090),
+        ]
```

Comment:
As mentioned in test data, the values it should return for semester are A, B and C.

This will work as an initial test based on the assumption that the migration data is the only data in the system. However, a test module shouldn't assume that specific data is in the system unless that is part of the production system that it is testing. For the test module to be stand-alone, the data that you are verifying should be added in the test module itself, and should use a filter so you get the specific data you are looking for instead of using .all

Reply:
@simzacks You are right, that was my intention. I saw now that the choises didn't work as i wanted them to work. I saw in your presentation what you did in 'Gender' class and made changes at my class according to that.